### DISCUSSION TITLE
category with many to many relation slows down queries (10MB+ response size)

### DISCUSSION THREAD

**Discussion by immunity20** [strapi/strapi#7993] [Legacy Topics]
Hello, I have genres and movies collections with many to many relation between them.

Front end or back end queries to fetch movies or genres takes too long time and from request details as you can see it downloads 10MBytes+ of data

Also in front-end when i try to fetch only 10 movies i still get too many MB of data downloaded (the response json contains an array of 10 movies which have a property genres with id of all the movies (~ 10000 movies) so its 10*10000 and thats probably what cause the issue.

Is there a way to avoid that?
![category_problem](https://user-images.githubusercontent.com/17274326/93932926-9300a180-fd29-11ea-83ca-8b48daf1eebe.png)

**Comment by derrickmehaffy** [comment]
Can you please submit this as a bug report and not a discussion?

**Reply by immunity20** [reply]
yes of course

**Reply by derrickmehaffy** [reply]
Locking for https://github.com/strapi/strapi/issues/7994
